model/base_variables.py

```diff
@@ -15,15 +15,17 @@ def __init__(self, model, competencies, staff, time_set, shifts_set, off_shifts_
         self.sundays = time_set["sundays"]
 
         self.y = self.add_y()
+
         self.x = self.add_x()
         self.w = self.add_w()
         self.mu = self.add_mu()
         self.delta = self.add_delta()
         self.lam = self.add_lambda()
 
     def add_y(self):
+        y = {(c,e,t): 0 for c in self.competencies for e in self.employees for t in self.time_periods[c]}
         return self.model.addVars(
-            self.competencies, self.employees, self.time_periods, vtype=GRB.BINARY, name="y"
+            y, vtype=GRB.BINARY, name="y"
```

Comment:
Does Gurobi unpack these variables so that the resultat is a Gurobi y-variable with only keys for the relevant `(c, e, t)` combinations and not _every possible combination_?

Reply:
Yes. If we create dictionaries before adding the variables, Gurobi would only use the keys present in that dictionary instead of creating values for all combinations of keys created from the sets

Reply:
What I haven't realised before now is that we should probably use this to only create y's for employees that have the competency to cover said demand.

Reply:
That´s great! Makes the code more readable and robust. To get this PR merged ASAP, I suggest you create a task on Trello or somewhere, and fix this later, if you find it worthwile :)

Reply:
Changing to employees_with_competencies creates new bugs that would take som time figuring out. Will leave it like this until later.